DEX.md

```diff
@@ -0,0 +1,99 @@
+3. Platform providers: Owners of the DEX.
+
+
+When a trade occurs, the trader pays a fee. This fee usually consists of:
+1. A platform fee that goes to the DEX protocol, and
+2. An LP fee that is distributed to liquidity providers, proportional to the amount of liquidity they contributed.
+
+Some platforms allow LPs to claim and reinvest their fees, effectively compounding their rewards.
+
+# Market Ratio
+
+A liquidity pool maintains a market ratio (MR) between its two tokens:
+
+$MR = \frac{reserve_0}{reserve_1}$
+
+Equivalently, it is the price of $T_1$ quoted in $T_0$. Intuitively, the higher the relative abundance of $T_0$ compared to $T_1$, the cheaper $T_1$ becomes in terms of $T_0$.
+
+When an LP deposits $amount_0$ and $amount_1$ to the pool (deposit ratio $DR = \frac{amount_0}{amount_1}$), the pool takes the following amounts from $T_0$ and $T_1$ respectively:
+
+1. if $DR == MR$ => ($amount_0$, $amount_1$)
+2. if $DR >  MR$ => ($amount_1 \times MR$, $amount_1$)
+3. if $DR <  MR$ => ($amount_0$, $amount_0 \times MR$)
+
+This ensures that liquidity is always added proportional to the current reserves' balances.
+If $DR \ne MR$, the excess tokens are returned to the LP.
+
+# Trading
+
+Assume a trader wants to swap $T_0$ for $T_1$. To trade, they send $amount_0$ of $T_0$.
+
+In the Automated Market Maker (AMM) model popularized by Uniswap V2, liquidity is defined as:
+
+$L^2 = reserve_0 \times reserve_1$
+
+This is called the *Constant Product Invariant*. The key idea is that, when trades happen, the product of reserves remains unchanged:
+
+$reserve_0 \times reserve_1 = (reserve_0 + amount_0) \times (reserve_1 - amount_1)$
+
+While this is the most common AMM model, other variants exist (e.g., constant sum for stablecoins, hybrid curves like Curve’s StableSwap).
+The adaptor is, at this stage, implemented exclusively for Kongswap. Since Kongswap (as of 5th September 2025) employs the constant product AMM, our implementation and tests are aligned with that model.
+
+$amount_1$ is the amount of $T_1$ that is going to be removed from the pool. Before sending
+it to the user, some fees are deducted: a platform fee (paid to the DEX) and an LP fee (accrued to liquidity providers, proportional to their contribution).
+
+# Withdrawing Liquidity
+
+LPs can withdraw their tokens at any time. However, due to trades happening in between, they rarely receive back the same token amounts they initially deposited. Instead, they withdraw, depending on the net direction of trades:
+
+1. More of one token,
+2. Less of the other, and
+3. The fees they have accrued during trading activity.
+
+Some DEXes provide the LPs with the functionality to withdraw their accrued fee and, if desired, reinvest it to
+the protocol to compound their rewards.
+
+# Pitfalls and Fallacies
+
+In this section, we will highlight common pitfalls and misconceptions.
```

Comment:
```suggestion
In this section, we will highlight some important pitfalls and facts.
```

(If we write misconsepctions, it is no clear if what we state below is now the truth or a misconseption)